applications/ConvectionDiffusionApplication/custom_utilities/embedded_mls_constraint_process.h

```diff
@@ -0,0 +1,264 @@
+    {
+        const Parameters default_parameters = Parameters(R"({
+            "model_part_name" : "",
+            "mls_extension_operator_order" : 1,
+            "avoid_zero_distances" : true,
+            "deactivate_negative_elements" : true,
+            "deactivate_intersected_elements" : false
+        })" );
+
+        return default_parameters;
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const override
+    {
+        return "EmbeddedMLSConstraintProcess";
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "EmbeddedMLSConstraintProcess";
+    }
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart* mpModelPart = nullptr;
+
+    std::size_t mMLSExtensionOperatorOrder;
+
+    bool mAvoidZeroDistances;
+
+    bool mDeactivateNegativeElements;
+    bool mDeactivateIntersectedElements;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void CalculateConformingExtensionBasis(
+        NodesCloudMapType& rExtensionOperatorMap);
+
+    void ApplyExtensionConstraints(
+        NodesCloudMapType& rExtensionOperatorMap);
+
+    void SetInterfaceFlags();
+
+    void ReactivateElementsAndNodes();
+
+    void ModifyDistances();
+
+    bool IsSplit(const GeometryType& rGeometry);
+
+    bool IsNegative(const GeometryType& rGeometry);
+
+    MLSShapeFunctionsFunctionType GetMLSShapeFunctionsFunction();
+
+    void SetNegativeNodeSupportCloud(
+        const NodeType& rNegativeNode,
+        PointerVector<NodeType>& rCloudNodes,
+        Matrix& rCloudCoordinates);
+
+    double CalculateKernelRadius(
+        const Matrix& rCloudCoordinates,
+        const array_1d<double,3>& rOrigin);
+
+    std::size_t GetRequiredNumberOfPoints();
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Default constructor.
+    EmbeddedMLSConstraintProcess() = delete;

```

Comment:
Very minor, but I'd move this to the life cycle section for the sake of clarity.